// Copyright (c) YugaByte, Inc.
syntax = "proto2";

package yb.tserver;

option java_package = "org.yb.tserver";

import "yb/tserver/tserver.proto";

service TabletServerBackupService {
  rpc TabletSnapshotOp(TabletSnapshotOpRequestPB)
      returns (TabletSnapshotOpResponsePB);
}

message TabletSnapshotOpRequestPB {
  enum Operation {
    UNKNOWN = 0;
    CREATE = 1;
    RESTORE = 2;
    DELETE = 3;
  }

  // UUID of server this request is addressed to.
  optional bytes dest_uuid = 1;

  optional Operation operation = 2;

  optional bytes snapshot_id = 3;

  // Tablet to snapshot.
  optional bytes tablet_id = 4;

  optional fixed64 propagated_hybrid_time = 5;
}

message TabletSnapshotOpResponsePB {
  optional TabletServerErrorPB error = 1;

  optional fixed64 propagated_hybrid_time = 2;
}
